src/repository.py

```diff
@@ -0,0 +1,221 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Module for handling interactions with git repository."""
+
+import base64
+import logging
+import re
+from pathlib import Path
+
+from git import GitCommandError
+from git.repo import Repo
+from github import Github
+from github.GithubException import GithubException
+from github.Repository import Repository
+
+from .constants import DOCUMENTATION_FOLDER_NAME
+from .exceptions import InputError, RepositoryClientError
+
+GITHUB_HOSTNAME = "github.com"
+HTTPS_URL_PATTERN = re.compile(rf"^https?:\/\/.*@?{GITHUB_HOSTNAME}\/(.+\/.+?)(.git)?$")
+ACTIONS_USER_NAME = "upload-charms-docs-bot"
+ACTIONS_USER_EMAIL = "upload-charms-docs-bot@users.noreply.github.com"
+ACTIONS_PULL_REQUEST_TITLE = "[upload-charm-docs] Migrate charm docs"
+ACTIONS_PULL_REQUEST_BODY = (
+    "This pull request was autogenerated by upload-charm-docs to migrate "
+    "existing documentation from server to the git repository."
+)
+PR_LINK_NO_CHANGE = "<not created due to no changes in repository>"
+
+CONFIG_USER_SECTION_NAME = "user"
+CONFIG_USER_NAME = (CONFIG_USER_SECTION_NAME, "name")
+CONFIG_USER_EMAIL = (CONFIG_USER_SECTION_NAME, "email")
+
+
+class Client:
+    """Wrapper for git/git-server related functionalities."""
+
+    def __init__(self, repository: Repo, github_repository: Repository) -> None:
+        """Construct.
+
+        Args:
+            repository: Client for interacting with local git repository.
+            github_repository: Client for interacting with remote github repository.
+        """
+        self._git_repo = repository
```

Comment:
We could consider refactoring of this class, it was just moved to a different file in this PR because the class started to be useful not just for creating PRs. I propose that we make this out of scope for this PR

Reply:
Agreed